### PR TITLE
Issue #37, allow mixed inputs again

### PR DESCRIPTION
**Pull request by FPGro**
And also aim for higher performance.
TODO:
- ~~check behaviour of invalid input~~ fine
- ~~check `Base.convert`s again~~ are good as they are
- ~~check tests~~ related tests pass

### CODE REVIEW

**Comment by FPGro** [comment]
Cool, works surprisingly well:
```
Points: Test Failed at /home/runner/work/Meshes.jl/Meshes.jl/test/points.jl:75
  Expression: Point{1, Int}((2, 3))
    Expected: DimensionMismatch
      Thrown: MethodError
```
This is actually fine, just a different error type because I don't catch it manually now but let `SVector` do the work.

**Comment by juliohm** [comment]
That is awesome. One of the errors in the CI is unrelated, trying to fix it in the master branch.

Printing single precision floats is incosistent:

```julia
julia> t = (1.0f0, 1.0f0)
(1.0f0, 1.0f0)

julia> println(stdout, t)
(1.0f0, 1.0f0)

julia> println(stdout, t[1])
1.0
```

Trying to figure out if I should open an issue in Julia.

**Comment by FPGro** [comment]
Uhm, the issue appears to be here: 
https://github.com/JuliaGeometry/Meshes.jl/blob/0b765970714167218ecc8c247ad7a888246d33b3/test/traits.jl#L27
and probably the line 2 below, too.
```
3 DummyDomain{2,Float32}
  └─Ball{2,Float32}(Point(1.0f0, 1.0f0), 1.0f0)
  └─Ball{2,Float32}(Point(2.0f0, 2.0f0), 1.0f0)
  └─Ball{2,Float32}(Point(3.0f0, 3.0f0), 1.0f0)
# vs:
3 DummyDomain{2,Float32}
  └─Ball{2, Float32}(Point(1.0f0, 1.0f0), 1.0f0)
  └─Ball{2, Float32}(Point(2.0f0, 2.0f0), 1.0f0)
  └─Ball{2, Float32}(Point(3.0f0, 3.0f0), 1.0f0)
```

I don't have the slightest clue how the constructors changed *that* behaviour. @juliohm can you look at this and maybe fix the test? :) 

PS: and maybe delete that disabled thing here, I forgot to clean it up, sorry ^^ I'm off for a couple of hours now
https://github.com/FPGro/Meshes.jl/blob/0d40bb827c43d7a9c4754448a90b1ce8ba7b6e21/src/points.jl#L34

**Comment by juliohm** [comment]
I meant that this error is not related to your work. It is a failure caused by the printing issue illustrated above. I introduced this test without knowing that single precision floats print as if they were double precision.

**Comment by FPGro** [comment]
> I meant that this error is not related to your work. It is a failure caused by the printing issue illustrated above. I introduced this test without knowing that single precision floats print as if they were double precision.

Yeah sorry I wrote my comment before I read yours. If that's just a printing issue, then it should be ready for review I guess, all other tests pass. :)

File: src/points.jl [hunk 1]
```diff
@@ -21,20 +21,18 @@ O = Point(0.0, 0.0) # origin of 2D Euclidean space
...
+Point{Dim,T}(coords...) where {Dim,T} = Point{Dim,T}(SVector{Dim,T}(coords...))
+Point(coords::SVector{Dim,T}) where {Dim,T} = Point{Dim,T}(coords)
+Point(coords::AbstractVector{T}) where {T} = Point{length(coords),T}(coords)
+Point(coords...) = Point(SVector(coords...))
 
 # catches mismatching tuple dimensions
-Point{Dim1,T}(::NTuple{Dim2}) where {T,Dim1,Dim2} = 
-  throw(DimensionMismatch("Can't construct a Point{$(Dim1),$(T)} with an input of length $(Dim2)"))
+#Point{Dim1,T}(::NTuple{Dim2}) where {T,Dim1,Dim2} = 
+#  throw(DimensionMismatch("Can't construct a Point{$(Dim1),$(T)} with an input of length $(Dim2)"))
```
Comment (juliohm) [inline src/points.jl]: Should this be removed as opposed to commented out?

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/points.jl (hunk 1 above)
Comment (FPGro) [reply src/points.jl]: Yes I forgot to clean up ^^ wait a second

**Comment by juliohm** [comment]
Should we add new tests to catch the mixed case? I understand that this PR now allows it, so probably add a few tests to make sure they are preserved in future changes.

**Comment by FPGro** [comment]
> 
> 
> Should we add new tests to catch the mixed case? I understand that this PR now allows it, so probably add a few tests to make sure they are preserved in future changes.

Sure, no problem

**Comment by FPGro** [comment]
Okay tests are in place, only the unrelated printing-test fails

**Review by juliohm — APPROVED** [review]
Amazing work as usual! ❤️ I will merge it given that the other error is completely unrelated and is being sorted out in master.

**Pull request merged**